ui/scripts/inlineHtml/inlineScripts.js

```diff
@@ -922,56 +927,50 @@ const updateCopiedAbilities = function () {
       }
       // Try to find an ability for each skill.
-      for (let skill of ['dodge', 'block', 'parry', 'aim spell', 'spell touch', 'engulf with spell']) {
+      for (let skill of ['guard', 'spell hit', 'affliction', 'mental']) {
         // First, get the ability assuming there is no training
-        let ability = DX_n + (skill === 'aim spell' ?
-          min_expertises['attack'] + 4 :
-          skill === 'spell touch' ?
-          min_expertises['attack'] + 5 :
-          skill === 'engulf with spell' ?
-          min_expertises['attack'] + 3 :
-          Math.max(
-            min_expertises['defense'],
-            min_expertises['defend']) - 3);
-        let skill_index = skill_map[skill];
-        if (isValueDefined(skill_index) && values[disciplineinfoFields[skill_index]] === 'D') {
-          skill_index = null;
-        }
-        // Check for the user using "shield" as an alternative for "block"
-        if (!isValueDefined(skill_index) && skill === 'block') {
-          skill_index = skill_map['shield'];
-          if (isValueDefined(skill_index) && values[disciplineinfoFields[skill_index]] === 'D') {
-            skill_index = null;
-          }
-        }
-        // Check for the user using 'engulf' as an alternative for 'engulf with spell'
-        if (!isValueDefined(skill_index) && skill === 'engulf with spell') {
-          skill_index = skill_map['engulf'];
-          if (isValueDefined(skill_index) && values[disciplineinfoFields[skill_index]] === 'D') {
-            skill_index = null;
+        let base = (skill === 'guard' ? DX_n - 3 :
+                    skill === 'spell hit' ? DX_n + 4 :
+                    skill === 'affliction' ? BR_n + 1 :
+                    IQ_n + 1)
+        let ability = base + (skill === 'guard' ?
+                              Math.max(min_focus_plus_expertises['defense'],
+                                       min_focus_plus_expertises['defend']):
+                              min_focus_plus_expertises['attack']);
+        // Check for both alternative names for the skill and obsolete names.
+        let possible_names = (skill === 'guard' ? ['guard', 'dodge', 'block', 'shield', 'parry'] :
+                              skill === 'spell hit' ? ['spell hit', 'engulf', 'engulf with spell',
+                                                       'aim spell', 'spell touch'] :
+                              skill === 'affliction' ? ['affliction', 'afflict', 'affliction attack'] :
+                              ['mental', 'mental attack']);
+        for (let name of possible_names) {
+          let skill_index = skill_map[name];
+          if (isValueDefined(skill_index) && values[disciplineinfoFields[skill_index]] !== 'D') {
+            possible_ability = Number(values[abilityFields[skill_index]]);
+            if (name === 'spell touch') {
+              possible_ability = possible_ability - 1; // Convert DX+5 to DX+4
```

Comment:
What is this doing? It seems like this will be a bit of a surprise?
Instead, we could do a data migration change that updates the `attr_skillbase` for the matching skill name as an `sheet:opened` call.

The value does show up here, so it does seem odd.
![image](https://github.com/user-attachments/assets/4a7fcd77-4915-45d6-90df-c49908d399a9)

Also - if we really want exactly `DX+4`, then why not just do `possibleAbility = 4;` to reduce risk of someone already having it at 4, then getting nerfed to 3.

Reply:
RIP Spell Touch, Aim Spell, and Engulf with Spell.
Now, there is just Spell Hit that works for all of those attacks. It is which is DX+4, but it gets a +1 if you are doing a Touch attack and a -1 if you are doing an area attack. This code is to make old characters that still use the old attacks be equivalent. In other words, if a character only had Spell Touch, it gets converted to the equivalent Spell Hit ability.